README.md

```diff
@@ -234,32 +276,41 @@ metadata:
   displayName: string # optional
 spec:
   thresholdMetric: # either thresholdMetric or ratioMetric should be provided
-    source: string # data source for the metric
-    queryType: string # a name for the type of query to run on the data source
-    query: string # the query to run to return the metric
-    metadata: # optional, allows data source specific details to be passed
+    metricSource:
+      name: string # optional, this field can be used to refer to DataSource object
```

Comment:
To align with existing usage in the spec wouldn't a reference to a `DataSource` object need a `metricSourceRef` field as opposed to utilizing `name`?

With this approach, it's not intuitive as to whether the `name` refers to a reference or is just a name. At least without looking across all possible YAML files

Reply:
Sure! I agree with that, let's keep our naming convention consistent across our specification

Reply:
Thanks for the change.

One question, is the expectation that if `metricSourceRef` is used the `type` would not be present on `metric` because the `type` would be on the `DataSource`?

If that's the expectation, should we mention it?

Reply:
Yup! We have mentioned that: https://github.com/OpenSLO/OpenSLO/blob/DataSource-as-separate-kind/README.md?plain=1#L436

Reply:
Apologies, I missed that! Thanks